pkg/assembler/backends/backends.go

```diff
@@ -26,57 +26,59 @@ import (
 // GraphQL interface and this is enforced by this interface.
 type Backend interface {
 	// Retrieval read-only queries for software trees
-	Packages(ctx context.Context, pkgSpec *model.PkgSpec) ([]*model.Package, error)
-	Sources(ctx context.Context, sourceSpec *model.SourceSpec) ([]*model.Source, error)
+	Artifacts(ctx context.Context, artifactSpec *model.ArtifactSpec) ([]*model.Artifact, error)
+	Builders(ctx context.Context, builderSpec *model.BuilderSpec) ([]*model.Builder, error)
 	Cve(ctx context.Context, cveSpec *model.CVESpec) ([]*model.Cve, error)
 	Ghsa(ctx context.Context, ghsaSpec *model.GHSASpec) ([]*model.Ghsa, error)
 	Osv(ctx context.Context, osvSpec *model.OSVSpec) ([]*model.Osv, error)
-	Artifacts(ctx context.Context, artifactSpec *model.ArtifactSpec) ([]*model.Artifact, error)
-	Builders(ctx context.Context, builderSpec *model.BuilderSpec) ([]*model.Builder, error)
+	Packages(ctx context.Context, pkgSpec *model.PkgSpec) ([]*model.Package, error)
+	Sources(ctx context.Context, sourceSpec *model.SourceSpec) ([]*model.Source, error)
 
 	// Retrieval read-only queries for evidence trees
-	HashEqual(ctx context.Context, hashEqualSpec *model.HashEqualSpec) ([]*model.HashEqual, error)
-	IsOccurrence(ctx context.Context, isOccurrenceSpec *model.IsOccurrenceSpec) ([]*model.IsOccurrence, error)
-	HasSBOM(ctx context.Context, hasSBOMSpec *model.HasSBOMSpec) ([]*model.HasSbom, error)
-	IsDependency(ctx context.Context, isDependencySpec *model.IsDependencySpec) ([]*model.IsDependency, error)
-	PkgEqual(ctx context.Context, pkgEqualSpec *model.PkgEqualSpec) ([]*model.PkgEqual, error)
-	HasSourceAt(ctx context.Context, hasSourceAtSpec *model.HasSourceAtSpec) ([]*model.HasSourceAt, error)
 	CertifyBad(ctx context.Context, certifyBadSpec *model.CertifyBadSpec) ([]*model.CertifyBad, error)
 	CertifyGood(ctx context.Context, certifyGoodSpec *model.CertifyGoodSpec) ([]*model.CertifyGood, error)
-	Scorecards(ctx context.Context, certifyScorecardSpec *model.CertifyScorecardSpec) ([]*model.CertifyScorecard, error)
-	CertifyVuln(ctx context.Context, certifyVulnSpec *model.CertifyVulnSpec) ([]*model.CertifyVuln, error)
-	IsVulnerability(ctx context.Context, isVulnerabilitySpec *model.IsVulnerabilitySpec) ([]*model.IsVulnerability, error)
+	CertifyNoKnownVuln(ctx context.Context, certifyNoKnownVulnSpec *model.CertifyNoKnownVulnSpec) ([]*model.NoKnownVuln, error)
 	CertifyVEXStatement(ctx context.Context, certifyVEXStatementSpec *model.CertifyVEXStatementSpec) ([]*model.CertifyVEXStatement, error)
+	CertifyVuln(ctx context.Context, certifyVulnSpec *model.CertifyVulnSpec) ([]*model.CertifyVuln, error)
```

Comment:
hmm, my interpretation would be a "NoVuln" being a type of vuln node and then CertifyVuln with point to that node? It seems fine either way, but i feel like this may help with check for counterfactual

Reply:
Agree. I was thinking similar to having a "NoVuln" node.

Reply:
Oh, right. I can do those changes.

I was thinking that the client might want to issue separate calls, but this is cleaner

Reply:
Should be fixed now